objects/src/block/header.rs

```diff
@@ -181,3 +184,54 @@ impl BlockHeader {
         Hasher::hash_elements(&elements)
     }
 }
+
+impl Serializable for BlockHeader {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let Self {
+            prev_hash,
+            block_num,
+            chain_root,
+            account_root,
+            nullifier_root,
+            note_root,
+            batch_root,
+            proof_hash,
+            version,
+            timestamp,
+            sub_hash,
+            hash,
+        } = self;
+
+        prev_hash.write_into(target);
+        block_num.write_into(target);
+        chain_root.write_into(target);
+        account_root.write_into(target);
+        nullifier_root.write_into(target);
+        note_root.write_into(target);
+        batch_root.write_into(target);
+        proof_hash.write_into(target);
+        version.write_into(target);
+        timestamp.write_into(target);
+        sub_hash.write_into(target);
+        hash.write_into(target);
```

Comment:
Without de-structuring it uses half the lines of code :)

```suggestion
        self.prev_hash.write_into(target);
        self.block_num.write_into(target);
        self.chain_root.write_into(target);
        self.account_root.write_into(target);
        self.nullifier_root.write_into(target);
        self.note_root.write_into(target);
        self.batch_root.write_into(target);
        self.proof_hash.write_into(target);
        self.version.write_into(target);
        self.timestamp.write_into(target);
        self.sub_hash.write_into(target);
        self.hash.write_into(target);
```

Reply:
Yes, definitely, but the code becomes fragile. If somebody added field to structure, but forgot to add it to serialization method, it would still compile.

Reply:
Humm, fair point, although I think that is covered by the deserialize. But most importantly, it should be covered by a unit test